packages/at_client/lib/src/encryption_service/abstract_atkey_encryption.dart

```diff
@@ -29,104 +27,76 @@ abstract class AbstractAtKeyEncryption implements AtKeyEncryption {
+  /// [getMyCopyOfSharedSymmetricKey]
+  /// - Calls [createMyCopyOfSharedSymmetricKey] if
+  ///   [getMyCopyOfSharedSymmetricKey] returns the empty string
+  /// - Calls [verifyTheirCopyOfSharedSymmetricKey]
+  /// - Doesn't actually encrypt the value, leaves that to the relevant
+  ///   subclass.
   @override
   Future<dynamic> encrypt(AtKey atKey, dynamic value,
       {bool storeSharedKeyEncryptedWithData = true}) async {
-    String sharedWithPublicKey = '';
-    String encryptedSharedKey = '';
-    // 1. Get AES Key from the local storage
-    _sharedKey = await getSharedKey(atKey);
-    // Fetch the encryption public key of the sharedWith atSign
-    try {
-      sharedWithPublicKey = await _getSharedWithPublicKey(atKey);
-    } on AtPublicKeyNotFoundException catch (e) {
-      e.stack(AtChainedException(
-          Intent.shareData, ExceptionScenario.encryptionFailed, e.message));
-      rethrow;
-    }
-    // If sharedKey is empty, then -
-    // Generate a new sharedKey
-    // Encrypt the sharedKey with sharedWith public key
+    _sharedKey = await getMyCopyOfSharedSymmetricKey(atKey);
+
     if (_sharedKey.isEmpty) {
-      // Generate sharedKey
-      _sharedKey = EncryptionUtil.generateAESKey();
-      // Encrypt shared key with public key of sharedWith atSign.
-      encryptedSharedKey =
-          EncryptionUtil.encryptKey(sharedKey, sharedWithPublicKey);
-      // Update the encrypted sharedKey to local secondary with TTR
-      await updateEncryptedSharedKeyToSecondary(atKey, encryptedSharedKey);
-      // Encrypt the sharedKey with currentAtSignPublicKey and store it for future use.
-      String? currentAtSignEncryptionPublicKey;
-      try {
-        currentAtSignEncryptionPublicKey = await _atClient
-            .getLocalSecondary()!
-            .getEncryptionPublicKey(atKey.sharedBy!);
-      } on KeyNotFoundException catch (e) {
-        e.stack(AtChainedException(
-            Intent.fetchEncryptionPublicKey,
-            ExceptionScenario.fetchEncryptionKeys,
-            'Failed to encrypt and store the sharedKey'));
-        rethrow;
-      }
-      var encryptedSharedKeyForCurrentAtSign = EncryptionUtil.encryptKey(
-          sharedKey, currentAtSignEncryptionPublicKey!);
-      _storeSharedKey(atKey, encryptedSharedKeyForCurrentAtSign);
-    }
-    // For the existing shared_key, the encryptedSharedKey has to be fetched
-    // from the local secondary
-    if (encryptedSharedKey.isNull) {
-      encryptedSharedKey =
-          EncryptionUtil.encryptKey(sharedKey, sharedWithPublicKey);
-    }
-    // Check if the encryptedSharedKey is synced to remote secondary
-    // If not synced, update the key to remote secondary directly
-    if (!(await isEncryptedSharedKeyInSync(atKey))) {
-      await updateEncryptedSharedKeyToSecondary(atKey, encryptedSharedKey,
-          secondary: _atClient.getRemoteSecondary());
+      _sharedKey = await createMyCopyOfSharedSymmetricKey(atKey);
```

Comment:
Created new method `createMyCopyOfSharedSymmetricKey` for readability of this (encrypt) method and clarity around the new logic